infrastructure/tari_script/src/script.rs

```diff
@@ -349,7 +349,14 @@ impl TariScript {
     fn handle_check_height(stack: &mut ExecutionStack, height: u64, block_height: u64) -> Result<(), ScriptError> {
         let height = i64::try_from(height)?;
         let block_height = i64::try_from(block_height)?;
-        let item = StackItem::Number(block_height - height);
+
+        // Due to the conversion of u64 into i64 which would fail above if they overflowed, these
+        // numbers should never enter a state where a `sub` could fail. As they'd both be within range and 0 or above.
+        // This differs from compare_height due to a stack number being used, which can be lower than 0
+        let item = match block_height.checked_sub(height) {
+            Some(num) => StackItem::Number(num),
+            None => return Err(ScriptError::CompareFailed),
```

Comment:
```suggestion
            None => return Err(ScriptError::StackUnderflow),
```
I would rather return the actual error here:

Reply:
At one point I had the same thought but the `StackUnderflow` isn't a general `Underflow` error. I believe `StackUnderflow` is a specific error to the Input Stack underflowing during script execution. Which isn't the error here.

We could create an alternative error though. Change `CompareFailed` to `Underflow` or `CompareFailed(message)` with a specific message.

Reply:
`CompareFailed(message)` is probably the best way forward. Compare failed lets you believe the comparison failed, not that its an underflow.

Reply:
I suggest `MathematicalUnderflow` ^^ to differentaiate from `StackUnderflow`

Reply:
Updated to `CompareFailed(message)`